packages/flutter_tools/lib/src/commands/build_windows.dart

```diff
@@ -52,10 +61,26 @@ class BuildWindowsCommand extends BuildSubCommand {
     if (!globals.platform.isWindows) {
       throwToolExit('"build windows" only supported on Windows hosts.');
     }
+
+    String? targetPlatformArg = stringArg('target-platform');
+    final String channel = globals.flutterVersion.channel;
+    final bool onStableOrBetaChannel = channel == 'stable' || channel == 'beta';
+    if ((targetPlatformArg != null) && onStableOrBetaChannel) {
+      throwToolExit('Option --target-platform can only be used on master channel.');
+    }
```

Comment:
Does cross compilation from x64 host to arm64 target work? If cross compilation doesn't work, should we remove this option for now?

Reply:
Yes it works in both ways (x64 -> arm64 and arm64 -> x64). However it works only for master channel (and user branch), thus this check. In the design document, we agreed on this behavior, which is what is implemented here.